@here/harp-map-controls/lib/CameraAnimationBuilder.ts

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright (C) 2017-2020 HERE Europe B.V.
+ * Licensed under Apache 2.0, see full license in LICENSE
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import { GeoCoordinates, MathUtils } from "@here/harp-geoutils";
+import { LookAtParams, MapView } from "@here/harp-mapview";
+import THREE = require("three");
+
+import { CameraKeyTrackAnimationOptions, ControlPoint } from "./CameraKeyTrackAnimation";
+
+/**
+ * A Utility class for the creation of Camera Animations
+ */
+export class CameraAnimationBuilder {
+    /**
+     * Get the current [[LookAtParams]] from [[MapView]]
+     *
+     * @param mapView The MapView
+     */
+    static getLookAtFromView(
+        mapView: MapView
+    ): Pick<LookAtParams, "target" | "tilt" | "heading" | "distance"> {
+        return {
+            target: mapView.target,
+            heading: mapView.heading,
+            tilt: mapView.tilt,
+            distance: mapView.targetDistance
+        };
+    }
+
+    /**
+     * Appends a [[ControlPoint]] to [[CameraKeyTrackAnimationOptions]]
+     *
+     * @param options The [[CameraKeyTrackAnimationOptions]] that the [[ControlPoint]] should be
+     * appended to
+     * @param point The [[ControlPoint]] to append.
+     * @param appendTime The time it should take from the former end of the animation to the
+     * appended [[ControlPoint]] in seconds, @default 10
+     */
+    static appendControlPoint(
+        options: CameraKeyTrackAnimationOptions,
+        point: ControlPoint,
+        appendTime?: number
+    ) {
+        appendTime = appendTime ?? 10;
+        const timestamp =
+            options.controlPoints[options.controlPoints.length - 1].timestamp + appendTime;
+        point.timestamp = timestamp;
+        options.controlPoints.push(point);
+    }
+
+    /**
+     * Adds a [[ControlPoint]] to the beginning of an [[CameraKeyTrackAnimationOptions]]
+     *
+     * @param options
+     * @param point
+     * @param prependTime The time the animation from the inserted key to the next should take,
+     *  in seconds, @default 10 seconds
+     */
+    static prependControlPoint(
+        options: CameraKeyTrackAnimationOptions,
+        point: ControlPoint,
+        prependTime?: number
+    ) {
+        prependTime = prependTime !== undefined ? prependTime : 10;
+        for (const controlPoint of options.controlPoints) {
+            controlPoint.timestamp += prependTime;
+        }
+        point.timestamp = 0;
+        options.controlPoints.unshift(point);
+    }
+
+    /**
+     *
+     * Creates Options for a Bow Animation from the start to the target [[ControlPoint]]
+     *
+     * @param mapView
+     * @param startControlPoint
+     * @param targetControlPoint
+     * @param altitude The maximal altitude the bow should have, defaults to twice the start to
+     * target distance
+     * @param duration The duration of the Animation in seconds, @default 10
+     */
+    static createBowFlyToOptions(
```

Comment:
Yes, its intentional there, and an adaption on the native implementation and used in the Tokyo App,  but I think there is way more options on how this could look or be configured, when and if there CameraKeyTrackAnimation gets used more, maybe we will find some more  useful AnimationOptions that might be nice to have in the Builder